src/main/scala/com/yotpo/metorikku/output/writers/instrumentation/InstrumentationOutputWriter.scala

```diff
@@ -13,59 +13,52 @@ class InstrumentationOutputWriter(props: Map[String, String],
                                   instrumentationFactory: InstrumentationFactory) extends Writer {
   @transient lazy val log: Logger = LogManager.getLogger(this.getClass)
 
-  val keyColumnProperty: Option[String] = Option(props).getOrElse(Map()).get("keyColumn")
+  val valueColumnProperty: Option[String] = Option(props).getOrElse(Map()).get("valueColumn")
   val timeColumnProperty: Option[String] = Option(props).getOrElse(Map()).get("timeColumn")
 
   override def write(dataFrame: DataFrame): Unit = {
     val columns = dataFrame.schema.fields.zipWithIndex
-    val indexOfKeyCol = keyColumnProperty.flatMap(col => Option(dataFrame.schema.fieldNames.indexOf(col)))
+    val indexOfValCol = valueColumnProperty.flatMap(col => Option(dataFrame.schema.fieldNames.indexOf(col)))
     val indexOfTimeCol = timeColumnProperty.flatMap(col => Option(dataFrame.schema.fieldNames.indexOf(col)))
 
     log.info(s"Starting to write Instrumentation of data frame: $dataFrameName on metric: $metricName")
     dataFrame.foreachPartition(p => {
       val client = instrumentationFactory.create()
 
+      // use last column if valueColumn is missing
+      val actualIndexOfValCol = indexOfValCol.getOrElse(columns.length - 1)
+      val actualIndexOfTimeCol = indexOfTimeCol.getOrElse(columns.length)
```

Comment:
Are you sure that's the default? It should be empty if not....

Reply:
if timeColumn is missing then it doesn't have an index. and its default value is set in 'getTime' function

Reply:
I don't think you need this variable at all